src/sections/pools/farms/position/FarmingPosition.utils.tsx

```diff
@@ -73,111 +73,130 @@ export const useAllUserDepositShare = () => {
-        liquidityOutResult = calculate_liquidity_out.apply(this, params)
-
-        const lrnaDp = BN_10.pow(lrnaMeta.data.decimals.toNumber() ?? 12)
-        const lrna =
-          lernaOutResult !== "-1" ? new BN(lernaOutResult).div(lrnaDp) : BN_NAN
-
-        const valueDp = BN_10.pow(meta.decimals.toNumber() ?? 12)
-        const value =
-          liquidityOutResult !== "-1"
-            ? new BN(liquidityOutResult).div(valueDp)
-            : BN_NAN
-
-        let valueDisplay = BN_NAN
+    const rows = positions.reduce(
+      (memo, position) => {
+        const { data: omnipoolBalance } =
+          omnipoolBalances.find(
+            (omnipoolBalance) =>
+              omnipoolBalance.data?.assetId.toString() ===
+              position.data?.assetId.toString(),
+          ) ?? {}
+
+        const omnipoolAsset = omnipoolAssets.data?.find(
+          (omnipoolAsset) =>
+            omnipoolAsset.id.toString() === position.data?.assetId.toString(),
+        )
+
+        const spotPrice = spotPrices.data?.find(
+          (spotPrice) =>
+            spotPrice?.tokenIn === position.data?.assetId.toString(),
+        )
+
+        const meta = metas.data?.find(
+          (meta) => meta.id === position.data?.assetId.toString(),
+        )
+
+        if (
+          omnipoolBalance &&
+          meta &&
+          omnipoolAsset?.data &&
+          position.data &&
+          lrnaMeta.data &&
+          spotPrice &&
+          lrnaSp.data
+        ) {
+          let lernaOutResult = "-1"
+          let liquidityOutResult = "-1"
+
+          const [nom, denom] =
+            position.data.price.map((n) => new BN(n.toString())) ?? []
+          const price = nom.div(denom)
+          const positionPrice = price.times(BN_10.pow(18))
+
+          const params: Parameters<typeof calculate_liquidity_out> = [
+            omnipoolBalance.balance.toString(),
+            omnipoolAsset.data.hubReserve.toString(),
+            omnipoolAsset.data.shares.toString(),
+            position.data.amount.toString(),
+            position.data.shares.toString(),
+            positionPrice.toFixed(0),
+            position.data.shares.toString(),
+            "0", // fee zero
+          ]
+
+          lernaOutResult = calculate_liquidity_lrna_out.apply(this, params)
+          liquidityOutResult = calculate_liquidity_out.apply(this, params)
+
+          const lrnaDp = BN_10.pow(lrnaMeta.data.decimals.toNumber() ?? 12)
+          const lrna =
+            lernaOutResult !== "-1"
+              ? new BN(lernaOutResult).div(lrnaDp)
+              : BN_NAN
+
+          const valueDp = BN_10.pow(meta.decimals.toNumber() ?? 12)
+          const value =
+            liquidityOutResult !== "-1"
+              ? new BN(liquidityOutResult).div(valueDp)
+              : BN_NAN
+
+          let valueDisplay = BN_NAN
+
+          if (liquidityOutResult !== "-1" && spotPrice) {
+            valueDisplay = value.times(spotPrice.spotPrice)
+
+            if (lrna.gt(0)) {
```

Comment:
lrna variable could be BN_NAN. Is it correct to check whether NaN is greater than zero?